### PR TITLE
patients can download test result

### LINKED ISSUE #3082: Patient results download
### Description
As we see more people using SimpleReport results for things like air travel, we want to make sure they have access to results offline. We also want to make sure that they have the most accurate version of their results, as the patient link results page doesn't include all the information (ordering provider, CLIA, etc) that is occasionally required by patients (i.e. when they're submitting insurance claims).

### Proposed solution 
We want to add a download link to the patient results experience that prints out the same form that facilities can currently print out. This has already been approved by CDC security. Patients would only be able to download their results once they successfully pass through the DOB login.

### Additional context
Once this is completed, @emmastephenson should notify Alex at Oswald's Pharmacy in IL about the change.

### PR DESCRIPTION
**Pull request by zdeveloper**
## Related Issue or Background Info

- fixes #3082

## Changes Proposed

- add `Download result` button to patient experience
- verify downloading test results. preferably on a mobile device

## Additional Information

- added a `/pxp/verify/v2 `to give the information necessary for the pdf

### Infrastructure
- [x] **Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!**

### Design
- [x] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [x] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [x] Any content changes (including new error messages) have been approved by content team

### Support
- [x] Any changes that might generate new support requests have been flagged to the support team
- [x] Any changes to support infrastructure have been demo'd to support team

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
  - [x] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [x] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [x] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [x] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: frontend/src/app/Settings/Facility/FacilityForm.test.tsx [hunk 1]
```diff
@@ -108,6 +108,9 @@ describe("FacilityForm", () => {
     suggestionIsCloseEnoughSpy = jest
       .spyOn(smartyStreets, "suggestionIsCloseEnough")
       .mockReturnValue(false);
+    jest
+      .spyOn(smartyStreets, "getZipCodeData")
+      .mockReturnValue(Promise.resolve(undefined));
```
Comment (zdeveloper) [inline frontend/src/app/Settings/Facility/FacilityForm.test.tsx:113]: fixes these errors
image

**Review by emmastephenson — APPROVED** [review]
LGTM! Thanks for addressing some of the test warnings too 👍

File: frontend/src/patientApp/PxpApiService.ts [hunk 2]
```diff
@@ -42,12 +43,48 @@ export type SelfRegistrationData = Omit<
   };
 };
 
+export type VerifyV2Response = {
```
Comment (nickclyde) [inline frontend/src/patientApp/PxpApiService.ts:46]: Nice - thanks for strict typing this

**Review by nickclyde — APPROVED** [review]
Looks great! Tested in test and works well 👍

**Pull request merged**